flink-core/src/main/java/org/apache/flink/core/fs/local/LocalDataOutputStream.java

```diff
@@ -35,6 +38,8 @@
     /** The file output stream used to write data. */
     private final FileOutputStream fos;
 
+    private volatile boolean isClosed = false;
```

Comment:
accessing a volatile field on these code paths is very expensive; in general output streams are not thread safe, so it's safe to use a non-volatile variable here

Reply:
Good point - I haven't considered that accessing this volatile field on each write call would be expensive 👍